chimedb/core/connectdb.py

```diff
@@ -448,15 +448,16 @@ def get_connection(self):
         return connection
 
     def get_peewee_database(self):
-        self.ensure_route_to_database()
-        host, port = self._host_port()
-        try:
-            self._database = MySQLDatabaseReconnect(
-                self._db, host=host, port=port, user=self._user, passwd=self._passwd
-            )
-        except None:
-            # TODO More descriptive here.
-            raise ConnectionError("Failed to connect to database.")
+        if self._database is None:
```

Comment:
This changes the behaviour here in an undesirable way I think. The previous behaviour would check that the tunnel is alive and attempt to fix it if not, but this new version does not as it just checks if the connection object exists. That's one of the more common reasons that connection will die.

Beyond adding that behaviour back in, it's probably worth testing the DB connection is alive too, although I guess provided the tunnel is alive the MySQLDatabaseReconnect object is essentially doing that.

Reply:
How about this?